File: slSig/Impressao/docs/XPSCardPrinterSDK74099/samples/vb/lamination/lamination.vb
''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
' Copyright (c) Datacard Corporation.  All Rights Reserved.
'
' XPS Driver SDK vb.net lamination sample.
''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
Imports System.Drawing.Printing
Imports System.IO
Imports System.Reflection
Imports dxp01sdk

Namespace Lamination

    Public Class Lamination

        Public Shared Function GetLongExeName() As String
            Dim longExeName As String = Path.GetDirectoryName(Assembly.GetEntryAssembly().Location)
            Return longExeName
        End Function

        Public Shared Sub Main(args As String())

            Dim commandLineOptions As CommandLineOptions = commandLineOptions.CreateFromArguments(args)
            commandLineOptions.Validate()

            Dim bidiSpl As BidiSplWrap = Nothing
            Dim printerJobID As Integer = 0

            Try
                bidiSpl = New BidiSplWrap()
                bidiSpl.BindDevice(commandLineOptions.printerName)

                Dim driverVersionXml As String = bidiSpl.GetPrinterData(strings.SDK_VERSION)
                Console.WriteLine(Environment.NewLine & "driver version: " & Util.ParseDriverVersionXML(driverVersionXml) & Environment.NewLine)

                Dim printerOptionsXML As String = bidiSpl.GetPrinterData(strings.PRINTER_OPTIONS2)
                Dim printerOptionsValues As PrinterOptionsValues = Util.ParsePrinterOptionsXML(printerOptionsXML)

                '  A printer may have both laminator station 1 and station 2 installed.
                '  In that case the option will be "L1, L2" so 'Contains' as opposed to 
                '  '=' must be used... 
                If LaminationActions.Actions.doesNotApply <> commandLineOptions.L1Action Then
                    If Not printerOptionsValues._laminator.Contains("L1") Then
                        Throw New Exception(commandLineOptions.printerName + " does not have a station 1 laminator installed.")
                    End If
                End If

                If LaminationActions.Actions.doesNotApply <> commandLineOptions.L2Action Then
                    If Not printerOptionsValues._laminator.Contains("L2") Then
                        Throw New Exception(commandLineOptions.printerName + " does not have a station 2 laminator installed.")
                    End If
                End If

                If "Ready" <> printerOptionsValues._printerStatus AndAlso "Busy" <> printerOptionsValues._printerStatus Then
                    Throw New Exception(commandLineOptions.printerName + " is not ready. status: " + printerOptionsValues._printerStatus)
                End If

                If commandLineOptions.jobCompletion OrElse (commandLineOptions.hopperID.Length > 0) OrElse (commandLineOptions.cardEjectSide.Length > 0) Then
                    Dim hopperID As String = "1"
                    Dim cardEjectSide As String = "default"
                    printerJobID = Util.StartJob(bidiSpl,
                                                 If((commandLineOptions.hopperID.Length > 0), commandLineOptions.hopperID, hopperID),
                                                 If((commandLineOptions.cardEjectSide.Length > 0), commandLineOptions.cardEjectSide, cardEjectSide))
                End If

                Dim printDocument As New SamplePrintDocument(commandLineOptions)
                printDocument.PrintController = New StandardPrintController()
                AddHandler printDocument.BeginPrint, New PrintEventHandler(AddressOf printDocument.OnBeginPrint)
                AddHandler printDocument.PrintPage, New PrintPageEventHandler(AddressOf printDocument.OnPrintPage)
                printDocument.Print()

                If 0 <> printerJobID Then
                    ' wait for the print spooling to finish and then issue an EndJob():
                    Util.WaitForWindowsJobID(bidiSpl, commandLineOptions.printerName)
                    bidiSpl.SetPrinterData(strings.ENDJOB)
                End If

                If commandLineOptions.jobCompletion Then
                    Util.PollForJobCompletion(bidiSpl, printerJobID)
                End If
            Catch e As BidiException
                Console.WriteLine(e.Message)
                Util.CancelJob(bidiSpl, e.PrinterJobID, e.ErrorCode)
            Catch e As Exception
                Console.WriteLine(e.Message)
                If 0 <> printerJobID Then
                    Util.CancelJob(bidiSpl, printerJobID, 0)
                End If
            Finally
                bidiSpl.UnbindDevice()
            End Try
        End Sub
    End Class
End Namespace

File: slSig/Impressao/docs/XPSCardPrinterSDK74099/samples/csharp/lamination/lamination.cs
////////////////////////////////////////////////////////////////////////////////
// Copyright (c) Datacard Corporation.  All Rights Reserved.
//
// XPS Driver SDK csharp lamination sample.
////////////////////////////////////////////////////////////////////////////////
using System;
using System.Drawing.Printing;
using System.IO;
using System.Reflection;
using dxp01sdk;

namespace Lamination {

    public class Lamination {

        public static string GetLongExeName() {
            string longExeName = Path.GetDirectoryName(Assembly.GetEntryAssembly().Location);
            return longExeName;
        }

        private static void Main(string[] args) {
            CommandLineOptions commandLineOptions = CommandLineOptions.CreateFromArguments(args);
            commandLineOptions.Validate();

            BidiSplWrap bidiSpl = null;
            int printerJobID = 0;

            try {
                bidiSpl = new BidiSplWrap();
                bidiSpl.BindDevice(commandLineOptions.printerName);

                string driverVersionXml = bidiSpl.GetPrinterData(strings.SDK_VERSION);
                Console.WriteLine(Environment.NewLine + "driver version: " + Util.ParseDriverVersionXML(driverVersionXml) + Environment.NewLine);

                string printerOptionsXML = bidiSpl.GetPrinterData(strings.PRINTER_OPTIONS2);
                PrinterOptionsValues printerOptionsValues = Util.ParsePrinterOptionsXML(printerOptionsXML);

                //  A printer may have both laminator station 1 and station 2 installed.
                //  In that case the option will be "L1, L2" so 'Contains' as opposed to
                //  '==' must be used...
                if (LaminationActions.Actions.doesNotApply != commandLineOptions.L1Action) {
                    if (!printerOptionsValues._laminator.Contains("L1")) {
                        throw new Exception(commandLineOptions.printerName + " does not have a station 1 laminator installed.");
                    }
                }

                if (LaminationActions.Actions.doesNotApply != commandLineOptions.L2Action) {
                    if (!printerOptionsValues._laminator.Contains("L2")) {
                        throw new Exception(commandLineOptions.printerName + " does not have a station 2 laminator installed.");
                    }
                }

                if ("Ready" != printerOptionsValues._printerStatus && "Busy" != printerOptionsValues._printerStatus) {
                    throw new Exception(commandLineOptions.printerName + " is not ready. status: " + printerOptionsValues._printerStatus);
                }

                if (commandLineOptions.jobCompletion ||
                    (commandLineOptions.hopperID.Length > 0) ||
                    (commandLineOptions.cardEjectSide.Length > 0)) {
                    string hopperID = "1";
                    string cardEjectSide = "default";

                    printerJobID = Util.StartJob(
                        bidiSpl,
                        (commandLineOptions.hopperID.Length > 0) ? commandLineOptions.hopperID : hopperID,
                        (commandLineOptions.cardEjectSide.Length > 0) ? commandLineOptions.cardEjectSide : cardEjectSide);
                }


                SamplePrintDocument printDocument = new SamplePrintDocument(commandLineOptions);
                printDocument.PrintController = new StandardPrintController();
                printDocument.BeginPrint += new PrintEventHandler(printDocument.OnBeginPrint);
                printDocument.PrintPage += new PrintPageEventHandler(printDocument.OnPrintPage);
                printDocument.Print();
                printDocument.RestoreUserPreferences();

                if (0 != printerJobID) {
                    // wait for the print spooling to finish and then issue an EndJob():
                    Util.WaitForWindowsJobID(bidiSpl, commandLineOptions.printerName);
                    bidiSpl.SetPrinterData(strings.ENDJOB);
                }

                if (commandLineOptions.jobCompletion) {
                    Util.PollForJobCompletion(bidiSpl, printerJobID);
                }
            }
            catch (BidiException e) {
                Console.WriteLine(e.Message);
                Util.CancelJob(bidiSpl, e.PrinterJobID, e.ErrorCode);
            }
            catch (Exception e) {
                Console.WriteLine(e.Message);
                if (0 != printerJobID) {
                    Util.CancelJob(bidiSpl, printerJobID, 0);
                }
            }
            finally {
                bidiSpl.UnbindDevice();
            }
        }
    }
}